### PR TITLE
Add recipe for SuchTree

### PR DESCRIPTION
**Pull request by ryneches**
Add a recipe for SuchTree 1.2.

SuchTree is a Python library built with Cython for fast, thread-safe computations with phylogenetic trees. It is particularly suitable for MCMC sampling of distances and relationships within very large trees (up to millions of taxa). 

- PiPy : https://pypi.org/project/SuchTree/
- Development : https://github.com/ryneches/SuchTree/
- Issues : https://github.com/ryneches/SuchTree/issues
- Documentation : https://github.com/ryneches/SuchTree/blob/master/README.md
- Publication : https://joss.theoj.org/papers/10.21105/joss.00678

### CODE REVIEW

File: recipes/suchtree/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+source:
+  url: "https://files.pythonhosted.org/packages/db/f0/75245602f31f80d18a7ffd4b4739e2eb442e46801c1c0ea430c0f002f2e7/suchtree-1.2.tar.gz"
+  sha256: c45af37beb1a210097151b807ccde9966c72a3cfa99ecfe80251bcc0a14f1f60
+
+build:
+  number: 0
+  dependencies:
+    - cython
+    - scipy
+    - numpy >=2.0
```
Comment (bgruening) [inline recipes/suchtree/meta.yaml:17]:
```suggestion
```

File: recipes/suchtree/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  run_exports:
+    - {{ pin_subpackage( "suchtree", max_pin="x.x" ) }}
+  script: "{{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - cython
+    - scipy
+    - numpy >=2.0
```
Comment (bgruening) [inline recipes/suchtree/meta.yaml:27]:
```suggestion
```

File: recipes/suchtree/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    - pandas >=2.1
+    - numpy >=2.0
+    - scipy
+
+test:
+  imports:
+    - SuchTree
+
+about:
+  home: The package home page
```
Comment (bgruening) [inline recipes/suchtree/meta.yaml:47]: Can you please update this one?

File: recipes/suchtree/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{% set name = "SuchTree" %}
+{% set version = "1.2" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://files.pythonhosted.org/packages/db/f0/75245602f31f80d18a7ffd4b4739e2eb442e46801c1c0ea430c0f002f2e7/suchtree-1.2.tar.gz"
```
Comment (bgruening) [inline recipes/suchtree/meta.yaml:9]: can you please use the other pypi URL and include the version jinja variable?

File: recipes/suchtree/meta.yaml (hunk 3 above)
Comment (ryneches) [reply recipes/suchtree/meta.yaml]: Oh. Ah, whoops!

File: recipes/suchtree/meta.yaml (hunk 4 above)
Comment (ryneches) [reply recipes/suchtree/meta.yaml]: I'm once again regretting including capital letters.

**Review by martin-g — APPROVED** [review]

**Pull request merged**